### PR TITLE
Use flags instead TYPE_ enum in visual shaders editor

### PR DESCRIPTION
**Pull request by Chaosus**
This is just a small internal change - if we are going to create new shader modes then its better to use flags to combine them rather then checking enum.

### CODE REVIEW

File: editor/plugins/visual_shader_editor_plugin.cpp [hunk 1]
```diff
@@ -186,20 +186,20 @@ bool VisualShaderEditor::_is_available(int p_mode) {
 
 		int temp_mode = 0;
 
-		if (p_mode & VisualShader::TYPE_FRAGMENT) {
+		if (p_mode & TYPE_FLAGS_VERTEX) {
+			temp_mode |= 1;
```
Comment (akien-mga) [inline editor/plugins/visual_shader_editor_plugin.cpp:190]: I'm not familiar with the big picture here, but doesn't all this code just end up copying `p_mode` in `temp_mode`?

Comment (Chaosus) [reply editor/plugins/visual_shader_editor_plugin.cpp:190]: Ah, really, Sorry for the sooner merge, I will remove that in the later commits.

Comment (akien-mga) [reply editor/plugins/visual_shader_editor_plugin.cpp:190]: I didn't check if that's true though, it's just an intuition from reading the diff.

Comment (Chaosus) [reply editor/plugins/visual_shader_editor_plugin.cpp:190]: It's true - I've checked :)

**Pull request merged**